timemachine/cpp/src/kernels/k_nonbonded.cuh

```diff
@@ -52,46 +52,52 @@ void __global__ k_check_rebuild_coords_and_box_gather(
     }
 }
 
-template <typename RealType>
+template <typename RealType, int COORDS_DIM, int PARAMS_DIM>
 void __global__ k_gather_coords_and_params(
     const int N,
     const unsigned int *__restrict__ idxs,
     const RealType *__restrict__ coords,
     const RealType *__restrict__ params,
     RealType *__restrict__ gathered_coords,
     RealType *__restrict__ gathered_params) {
-
+    static_assert(COORDS_DIM == 3);
+    static_assert(PARAMS_DIM == PARAMS_PER_ATOM);
```

Comment:
What is the reasoning for making `COORDS_DIM` and `PARAMS_DIM` template parameters, given that they are fixed to these values?

Reply:
No strong preference either; I think the consistency argument makes sense.